worker-store-in-memory/src/main/java/com/hpe/caf/worker/datastore/inmemory/InMemoryDataStore.java

```diff
@@ -0,0 +1,211 @@
+
+    /**
+     * Remove the asset identified by the reference.
+     * @param reference a complete reference to be interpreted by the DataStore implementation
+     * @throws DataStoreException if data store cannot service the request.
+     */
+    @Override
+    public void delete(String reference) throws DataStoreException
+    {
+        numDeleteRequests.incrementAndGet();
+        dataMap.remove(reference);
+    }
+
+    /**
+     * Provide a stream to get data with the key reference.
+     * @param reference a complete reference to be interpreted by the DataStore implementation
+     * @return
+     * @throws DataStoreException
+     */
+    @Override
+    public InputStream retrieve(String reference) throws DataStoreException
+    {
+        numRetrieveRequests.incrementAndGet();
+        if(!dataMap.containsKey(reference)){
+            errors.incrementAndGet();
+            throw new DataStoreException("Failed to retrieve data, the reference does not exist.");
+        }
+        return new ByteArrayInputStream(dataMap.get(reference));
+    }
+
+    /**
+     * Return the byte size of the data at the reference.
+     * @param reference a complete reference to be interpreted by the DataStore implementation
+     * @return the byte size
+     * @throws DataStoreException
+     */
+    @Override
+    public long size(String reference) throws DataStoreException
+    {
+        return dataMap.get(reference).length;
+    }
+
+    /**
+     * Store data from a stream, in the key given by the partial reference.
+     * @param dataStream the stream of data which will be read and stored in the in memory data store.
+     * @param partialReference the partial reference, which the data will be stored relative to.
+     * @return absolute reference to the stored data which can be used to retrieve.
+     * @throws DataStoreException if the data store cannot service the request.
+     */
+    @Override
+    public String store(InputStream dataStream, String partialReference) throws DataStoreException
+    {
+        try {
+            return store(IOUtils.toByteArray(dataStream), partialReference);
+        } catch (IOException e) {
+            errors.incrementAndGet();
+            throw new DataStoreException("Could not store input stream.", e);
+        }
+    }
+
+    /**
+     * Store data from a byte array, in the key given by the partial reference.
+     * @param data the raw byte data to store in the in-memory data store.
+     * @param partialReference the partial reference, which the data will be stored relative to.
+     * @return absolute reference to the stored data which can be used to retrieve.
+     * @throws DataStoreException if the data store cannot service the request.
+     */
+    @Override
+    public String store(byte[] data, String partialReference) throws DataStoreException
+    {
+        numStoreRequests.incrementAndGet();
+        String absoluteReference = UUID.randomUUID().toString() + partialReference;
```

Comment:
I'm assuming we need to add another part to the reference so that the references are different - forming the absoluteReference? - we use the same partialReference which is the container ID?